### PR TITLE
Gutenboarding: convert design screenshots to webp

### PR DESCRIPTION
**Pull request by ramonjd**
## Changes proposed in this Pull Request

To use `webp` format for images in the Gutenboarding design selector.

Discussion in p1594723857059700-slack-CNMU1UHHB

## Testing instructions

Take a look at the screenshots at the `/new/design` step. 

How do they look?

Now check in IE11. Do the JPEGs load instead?

### CODE REVIEW

File: client/landing/gutenboarding/available-designs.ts [hunk 1]
```diff
@@ -16,14 +16,28 @@ interface AvailableDesigns {
...
 	// We temporarily show pre-generated screenshots until we can generate tall versions dynamically using mshots.
 	// See `bin/generate-gutenboarding-design-thumbnails.js` for generating screenshots.
 	// https://github.com/Automattic/mShots/issues/16
 	// https://github.com/Automattic/wp-calypso/issues/40564
 	if ( ! isEnabled( 'gutenboarding/mshot-preview' ) ) {
 		// When we update the static images, bump the version for cache busting
-		return `/calypso/page-templates/design-screenshots/${ design.slug }_${ design.template }_${ design.theme }.jpg?v=2`;
+		return `/calypso/page-templates/design-screenshots/${ design.slug }_${ design.template }_${
+			design.theme
+		}.${ canUseWebP ? 'webp' : 'jpeg' }?v=1`;
```
Comment (simison) [inline client/landing/gutenboarding/available-designs.ts:40]: Shouldn't extension here be jpg, not jpeg?

Comment (ramonjd) [reply client/landing/gutenboarding/available-designs.ts]: I don't know what you're talking about :trollface: 

P.S. _Thanks_ :)

File: client/landing/gutenboarding/available-designs.ts [hunk 2]
```diff
@@ -16,14 +16,28 @@ interface AvailableDesigns {
 
 const availableDesigns: Readonly< AvailableDesigns > = availableDesignsConfig;
 
+function getCanUseWebP() {
+	if ( typeof window !== 'undefined' ) {
+		const elem = document.createElement( 'canvas' );
+		if ( !! elem.getContext && elem.getContext( '2d' ) ) {
```
Comment (alshakero) [inline client/landing/gutenboarding/available-designs.ts:22]:
```suggestion
		if ( elem.getContext?.( '2d' ) ) {
```

File: client/landing/gutenboarding/available-designs.ts [hunk 3]
```diff
@@ -16,14 +16,28 @@ interface AvailableDesigns {
...
 	// We temporarily show pre-generated screenshots until we can generate tall versions dynamically using mshots.
 	// See `bin/generate-gutenboarding-design-thumbnails.js` for generating screenshots.
 	// https://github.com/Automattic/mShots/issues/16
 	// https://github.com/Automattic/wp-calypso/issues/40564
 	if ( ! isEnabled( 'gutenboarding/mshot-preview' ) ) {
 		// When we update the static images, bump the version for cache busting
-		return `/calypso/page-templates/design-screenshots/${ design.slug }_${ design.template }_${ design.theme }.jpg?v=2`;
+		return `/calypso/page-templates/design-screenshots/${ design.slug }_${ design.template }_${
+			design.theme
+		}.${ canUseWebP ? 'webp' : 'jpg' }?v=1`;
```
Comment (alshakero) [inline client/landing/gutenboarding/available-designs.ts:40]: the reverting of v=1 might cause some caching issues for Safari and IE users.

File: bin/generate-gutenboarding-design-thumbnails.js [hunk 4]
```diff
@@ -82,9 +82,27 @@ async function run() {
...
+					} )
+					.resize( outputWidth )
+					.webp() // default quality is 80
+					.toFile( `${ file }.webp` );
+			} );
+
+			image = await sharp( screenshot );
+			console.log( `Resizing and saving to ${ file }.jpg` );
+
+			await image.metadata().then( ( metadata ) => {
```
Comment (alshakero) [inline bin/generate-gutenboarding-design-thumbnails.js:105]: I would do 
```js
[`${ file }.webp`, `${ file }.jpg`].forEach(file => 
...
```

And keep one copy of this logic to make the code DRYer.

File: client/landing/gutenboarding/available-designs.ts (hunk 3 above)
Comment (ramonjd) [reply client/landing/gutenboarding/available-designs.ts:40]: Thanks for catching that! Fixing...

File: bin/generate-gutenboarding-design-thumbnails.js (hunk 4 above)
Comment (ramonjd) [reply bin/generate-gutenboarding-design-thumbnails.js:105]: Great tip. Appreciate it!! 🙇

**Comment by ramonjd** [comment]
This had to be reverted. The webp images didn't make it to production. Or at least nginx is reporting a 404

**Pull request merged**